script.js

```diff
@@ -1,4 +1,139 @@
+  }
+
+  // Next, let's choose the order of your dice rolled.
+  else if (currentGameMode == chooseordermode) {
+    // If player chooses 1st dice number as first numeral
+    myOutputValue =
+      "Your combined number is " +
+      diceRoll1 +
+      diceRoll2 +
+      ".<br> <br> Click submit to continue.";
+    var combinednumber = String(diceRoll1) + String(diceRoll2);
+
+    // If player chooses 2nd dice number as first numeral
+    if (input == diceRoll2) {
+      myOutputValue =
+        "Your combined number is " +
+        diceRoll2 +
+        diceRoll1 +
+        ".<br> <br> Click submit to continue.";
+      combinednumber = String(diceRoll2) + String(diceRoll1);
+    }
+
+    // let's push the combined number of the player into our database
+    playernumber.push(combinednumber);
+    console.log("player numbers array: " + playernumber);
+    console.log("output: " + myOutputValue);
+    console.log("combined number: " + combinednumber);
+    // let's return the game mode to roll dice mode.
+    currentGameMode = rolldicemode;
+    // don't forget to reset the database of diceNumberRolled, because this will affect the value of the combined number which is made of up DiceRoll1 and DiceRoll2.
+    diceNumbersRolled = [];
+
+    // This is how we restart the round for player 2
+    if (currentplayer == 2) {
+      // change game mode to decide which is winner
+      currentGameMode = "Announce Winner";
+    } else {
+      currentplayer = currentplayer + 1;
+    }
+  } else if ((currentGameMode = "Announce Winner")) {
+    myOutputValue = announceWinner();
+  }
   return myOutputValue;
 };
+//
+//
+//
+//
+//
+//
+var rollDiceGame = function (gamecounter) {
+  console.log("player switch" + currentplayer);
+  // First we need to make sure the dice is rolled twice.
+  while (gamecounter < 2) {
+    gamecounter = gamecounter + 1;
+    //Execute the roll dice function
+    var diceNumber = diceRoll();
+    //Store the dice numbers rolled
+    diceNumbersRolled.push(diceNumber);
+    console.log("push " + diceNumber);
+  }
+  console.log("dice number data store: " + diceNumbersRolled);
+
+  // Let's assign the dice numbers rolled to the global variables we created above.
+  diceRoll1 = diceNumbersRolled[0];
+  diceRoll2 = diceNumbersRolled[1];
+  console.log("1st dice number: " + diceRoll1);
+  console.log("2nd dice number: " + diceRoll2);
+  return (
+    "Alright Player " +
+    currentplayer +
+    "! <br> <br> You rolled " +
+    diceRoll1 +
+    " and " +
+    diceRoll2 +
+    ". <br> <br> Choose which number you want as the first numeral of your combined number."
+  );
+};
+//
+//
+//
+//
+//
+//
+var announceWinner = function () {
+  myOutputValue = "Player 1 wins! To play again, click submit.";
+  if (playernumber[1] > playernumber[0]) {
+    myOutputValue = "Player 2 wins! To play again, click submit.";
+  }
+  currentGameMode = rolldicemode;
+  currentplayer = 1;
+  playernumber = [];
+  return myOutputValue;
+};
```

Comment:
Per your question about:

> How do I format the code so that when its player 2, I can change the output message to say "Click submit to see who wins"?

This is the place to do it! The logic seems to be fine, and you are performing your resets here perfectly too.

Only small thing I will add, do you want to consider draws? If so:
```
var announceWinner = function () {
  // using 'var output' here instead of 'myOutputValue', if
  // you are changing `myOutputValue` to be scoped to
  // the `main` function instead of leaving it as a global
  // variable.
  var output = "Player 1 wins! To play again, click submit.";
  if (playernumber[1] > playernumber[0]) {
    output = "Player 2 wins! To play again, click submit.";
  } else if (playernumber[1] == playernumber[0]) {
    output = "It's a draw! To play again, click submit.";
  }
  currentGameMode = rolldicemode;
  currentplayer = 1;
  playernumber = [];
  return output;
};
```
Otherwise, do you have any particular concerns about the way you are setting up the `"... To play again, click submit."` copywriting?